sympy/sets/sets.py

```diff
@@ -842,6 +842,8 @@ def __new__(cls, start, end, left_open=False, right_open=False):
             left_open = true
         if end == S.Infinity:
             right_open = true
+        if start == S.Infinity and end != S.Infinity:
```

Comment:
As far as I can see, the second condition is not really necessary, since `Interval(oo, oo)` also results in an empty set anyway. Suggestion:
```
if start == S.Infinity or end == S.NegativeInfinity:
```

Reply:
True. Changing it.